x-pack/plugins/security/server/authorization/roles/elasticsearch_role.ts

```diff
@@ -352,3 +366,83 @@ export const compareRolesByName = (roleA: Role, roleB: Role) => {
 
   return 0;
 };
+
+interface DeserializeFeaturePrivilegesOptions {
+  features: KibanaFeature[];
+  serializedPrivileges: string[];
+  subFeaturePrivilegeIterator: SubFeaturePrivilegeIterator;
+  replaceDeprecatedKibanaPrivileges?: boolean;
+}
+
+function deserializeKibanaFeaturePrivileges({
```

Comment:
__note:__ this is the core of this PR - the place where we map deprecated privileges to non-deprecated ones.